calico-vpp-agent/watchers/peers_watcher.go

```diff
@@ -178,6 +188,48 @@ func (w *PeerWatcher) WatchBGPPeers(t *tomb.Tomb) error {
 					w.nodeStatesByName[new.Name] = *new
 				}
 				w.log.Debugf("Nodes updated, reevaluating peerings old %v new %v", old, new)
+			case common.BGPSecretChanged:
+				old, _ := evt.Old.(*v1.Secret)
+				new, _ := evt.New.(*v1.Secret)
+				secretEvt := ""
+				secretName := ""
+				// secret added
+				if old == nil && new != nil {
+					secretEvt = "add"
+					secretName = new.Name
+					w.log.Infof("New secret '%s' added", new.Name)
+				}
+				// secret deleted
+				if old != nil && new == nil {
+					secretEvt = "del"
+					secretName = old.Name
+					w.log.Infof("secret '%s' deleted", old.Name)
+				}
+				// secret updated
+				if old != nil && new != nil {
+					secretEvt = "upd"
+					secretName = old.Name
+					w.log.Infof("secret '%s' updated", old.Name)
+				}
+				// sweep through the peers and update the SecretChanged field of impacted peers
+				for _, peer := range state {
+					switch secretEvt {
+					case "add":
+						// Note: any future add event specifc processing code goes here. For now we fallthrough.
+						fallthrough
+					case "del":
+						// Note: any future delete event specifc processing code goes here. For now we fallthrough.
+						fallthrough
+					case "upd":
+						// BGP password has changed
+						if w.getSecretName(peer.BGPPeerSpec) == secretName {
```

Comment:
Couldn't we update the `Peer` object here ? (instead of setting `SecretChanged` to true )

Reply:
I didn't quite understand. You mean the `peer.BGPPeerSpec.Password` field?